actix-router/src/resource.rs

```diff
@@ -294,8 +294,12 @@ impl ResourceDef {
                 } else {
                     return false;
                 }
-                for idx in 0..idx {
-                    path.add(names[idx], segments[idx]);
+                for (idx, segment) in segments
+                    .iter_mut()
+                    .enumerate()
+                    .take_while(|(i, _)| *i != idx)
+                {
+                    path.add(names[idx], segment.take().unwrap());
                 }
```

Comment:
For functional style:
```suggestion
                segments
                    .iter_mut()
                    .flat_map(Option::take)
                    .zip(names)
                    .for_each(|(segment, name)| path.add(name, segment));
```
 for  procedural:
```suggestion
                for idx in 0..names.len() {
                    path.add(names[idx], segments[idx].take().unwrap());
                }
```

I find the procedural more readable here.
In both cases. `idx` is not used!

Reply:
Good point, it short circuits on failure, so we can just use all the names here.